packages/shared/src/schemas.ts

```diff
@@ -10,12 +15,30 @@ export const quoteQuerySchema = z.object({
 
 export type QuoteQueryParams = z.infer<typeof quoteQuerySchema>;
 
+export type ByteString = string | `0x${string}`;
+export interface CcmMetadata {
+  gas_budget: `0x${string}` | number;
+  message: ByteString;
+  source_address: 0; // tb removed
+  cf_parameters?: ByteString;
+}
```

Comment:
not sure if this works

```suggestion
export type CcmMetadata = z.infer<typeof ccmMetadataSchema>;
```